backend/app/api/api_v1/endpoints/domains.py

```
@@ -2193,6 +2230,87 @@ async def get_domain_stats(
     )
 
 
+@router.get("/{domain_id}/migration/readiness", response_model=MigrationReadinessResponse)
+async def get_domain_migration_readiness(
+    domain_id: str = Path(..., title="The domain ID or name"),
+    refresh: bool = Query(False, title="Refresh cached DNS result"),
+    db: Session = Depends(get_db),
```

Ruff (0.15.20)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)